l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -601,68 +601,82 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
     ) external payable override onlyAssetRouter onlyL1 returns (bytes memory bridgehubMintData) {
         require(whitelistedSettlementLayers[_settlementChainId], "BH: SL not whitelisted");
 
-        (uint256 _chainId, bytes memory _stmData, bytes memory _chainData) = abi.decode(_data, (uint256, bytes, bytes));
-        require(_assetId == stmAssetIdFromChainId(_chainId), "BH: assetInfo 1");
-        require(settlementLayer[_chainId] == block.chainid, "BH: not current SL");
-        settlementLayer[_chainId] = _settlementChainId;
+        BridgehubSTMAssetData memory bridgeData = abi.decode(_data, (BridgehubSTMAssetData));
```

Comment:
I would suggest to have different structs here and for `bridgeMint`. Even if they contain same number/types of fields, they are inherently different. The mint one contains data that STM/chain returned to them. Burn contains the data that was provided by the user (admin of the chain).

Also, this bridgehub function should be (if possible) generic over the implementation details of STMs/chains, i.e. it should never parse the inner `bytes` fields, such as `bridgeData.chainData`